mpcstats_lib.py

```diff
@@ -38,12 +38,33 @@ def print_data(data: Matrix):
             print_ln("data[{}][{}]: %s".format(i, j), data[i][j].reveal())
 
 
+def _mean(data: list[sint]) -> (float, int):
+    total = sum(if_else(i != MAGIC_NUMBER, i, 0) for i in data)
+    count = sum(if_else(i != MAGIC_NUMBER, 1, 0) for i in data)
+    return total / count, count
+
+
+def _variance(data: list[sint], use_bessels: bool) -> float:
+    # calculate mean of the data excluding magic numbers
+    mean, eff_size = _mean(data)
+
+    # replace magic number w/ mean to skip magic number in for loop below
+    data = Array.create_from(if_else(n != MAGIC_NUMBER, n, mean) for n in data)
+
+    eff_data_sum = sfloat(0)
+    for n in data:
+        eff_data_sum += (n - mean) ** 2
```

Comment:
Can you confirm if we should use `+=` in this case, or should we use `sfix.update`?
```suggestion
        eff_data_sum += (n - mean) ** 2
```

Reply:
It seems like we need to use `update` when we use `@for_range` decorator. In this case, I confirmed that both the current code and the code using `update` inside the loop body work. 

```python
    eff_data_sum = sfix(0)
    for n in data:
        eff_data_sum.update(eff_data_sum + (n - mean) ** 2) # this works
        #eff_data_sum += (n - mean) ** 2 # this works too
```

```python
    @for_range(len(data))
    def _(i):
        eff_data_sum.update(eff_data_sum + (data[i] - mean) ** 2)
        #eff_data_sum += (data[i] - mean) ** 2 # this doesn't compile
```

Reply:
I see. Thanks for confirming. So the conclusion is "if we're using a plain `for`, we can use += instead".